app/reducers/files.js

```diff
@@ -118,99 +94,60 @@ const filesReducer = handleActions({
     const { id } = payload;
     return state.set('currentFile', id);
   },
-  [FILE_UPDATE]: (state, { payload }) => {
-    const { text } = payload;
-    const id = state.get('currentFile', uuid());
-    let file = state.getIn(['files', id], Map({}));
-    file = file.merge({
-      contents: text || '',
-      changed: (file.get('changed') || file.get('contents') !== text)
-    });
+  [FILE_UPDATE]: (state, action) => {
+    const id = state.get('currentFile');
+    let file = state.getIn(['files', id]);
+    file = fileReducer(file, action);
     return state.setIn(['files', id], file);
```

Comment:
I see a couple instances of pulling something out of state, changing it, then setting it back into the state. I might be more concise to make use of `.update`. Something like... ```return state.updateIn(['files', id], file => fileReducer(file, action))```